### PR TITLE
Modifications

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: octomap_server/launch/octomap_mapping.launch [hunk 1]
```diff
@@ -8,16 +8,19 @@
 -->
 <launch>
```
Comment (agdespopoulos) [inline octomap_server/launch/octomap_mapping.launch]: I think we should revert this and set these values ourselves in our launch files.

File: octomap_server/src/OctomapServer.cpp [hunk 2]
```diff
@@ -64,7 +67,7 @@ OctomapServer::OctomapServer(ros::NodeHandle private_nh_)
   m_occupancyMinZ(-std::numeric_limits<double>::max()),
   m_occupancyMaxZ(std::numeric_limits<double>::max()),
   m_minSizeX(0.0), m_minSizeY(0.0),
-  m_filterSpeckles(false), m_filterGroundPlane(false),
+  m_filterSpeckles(false), m_filterGroundPlane(false), m_simpleGroundFilter(true),
```
Comment (agdespopoulos) [inline octomap_server/src/OctomapServer.cpp]:
```suggestion
  m_filterSpeckles(false), m_filterGroundPlane(false), m_simpleGroundFilter(false),
```

I think we should make it act as much like the upstream `octomap_server` by default as possible. We can set this to true when we launch.

**Review by agdespopoulos — COMMENTED** [review]
A couple questions - 

How much overlap is there between this and OctomapServerMultilayer (disregarding that the multilayer server's cross sections are hardcoded in)?

What's the purpose of turning off and on the map making switch (`m_start_making_map`)?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+*.settings/
+*.catkin_tools
+
+
+# Generated by dynamic reconfigure
+*.cfgc
+/cfg/cpp/
+/cfg/*.py
+
+
```
Comment (brenden-gibbons) [inline .gitignore]: Usual complaint - bunch of extra empty spaces

File: octomap_server/launch/octomap_mapping.launch (hunk 1 above)
Comment (brenden-gibbons) [reply octomap_server/launch/octomap_mapping.launch]: Agree 100%

**Review by brenden-gibbons — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
How much overlap is there between this and OctomapServerMultilayer (disregarding that the multilayer server's cross sections are hardcoded in)?
Both my changes and OctomapServerMultilayer seem to be attempting to do the same thing. I will look into whether what I'd need to do to not hardcode everything and make OctomapServerMultilayer more general.

**Comment by ghost** [comment]
What's the purpose of turning off and on the map making switch (m_start_making_map)?

this is to make sure octomap isn't taking in points while utbot system is in entrance frame.

**Comment by agdespopoulos** [comment]
> this is to make sure octomap isn't taking in points while utbot system is in entrance frame.

Would it be better then to just filter the point cloud input by frame ID? Configure it so that certain frame IDs are never added to the map.

**Comment by agdespopoulos** [comment]
Actually, do we have to filter out any frames at all? If a transform exists from the point cloud frame to the mapping frame, that's data we can use, and if a transform doesn't exist, the point cloud already won't be processed (due to the tf message filter).

**Comment by ghost** [comment]
> Actually, do we have to filter out any frames at all? If a transform exists from the point cloud frame to the mapping frame, that's data we can use, and if a transform doesn't exist, the point cloud already won't be processed (due to the tf message filter).

Few things: tf message filters used in mapping allow both entrance and world frame points through.
Second: octomap is alway working in world frame, I don't have map and world frame be seperate
Third: octomap received points in original sensor frame(so me_1512_top and isa500_sensor frames) so that it can do some internal processing on it. 

But that's besides the point, I guess we can filter out non-world points in mapping. I will  get rid of switch.

**Comment by ghost** [comment]
What are your thoughts on octomap frame? Should I make another map frame or should I just leave it in world frame?

**Comment by agdespopoulos** [comment]
> tf message filters used in mapping allow both entrance and world frame points through.

Octomap only (I think) accepts point clouds from frames that have a transform to the map frame (in this case, `world`). So it wouldn't accept point clouds from `entrance` unless there was a transform from `entrance` to `world`. If that transform existed, I don't see why we wouldn't want to accept those point clouds.

> Second: octomap is alway working in world frame, I don't have map and world frame be seperate
>
> What are your thoughts on octomap frame? Should I make another map frame or should I just leave it in world frame?

I think that's fine - but I haven't really put much thought into it and I don't know what the drawbacks of either approach really are. But I think that's fine.

> Third: octomap received points in original sensor frame(so me_1512_top and isa500_sensor frames) so that it can do some internal processing on it.

Octomap or mapping? Regardless, I think my comment about frames and transforms from your first point still stands here - while we're in `entrance` frame, there won't be a transform from `me_1512_top` to `world` (right? I think), so octomap would ignore those clouds.

**Comment by brenden-gibbons** [comment]
I agree with the removal of the switch, tried this out a few times in sim - there are no issues if the switch is set to true immediately, so I definitely am all for removing it.

**Review by agdespopoulos — COMMENTED** [review]
I think I'm happy enough with this.

Although - quick question about the necessity of the simple ground filter. Simple ground filter removes points if z is zero +/- ground distance filter height - as the comment says, it envelopes it and removes only points with z approximately equaling zero. However, the unmodified octomap server can be configured (with `pointcloud_min_z`) to filter out points with a z value below a certain value, which we could set to zero or just above zero. Do we have a need for negative z value points? If not, I think we can just use the `pointcloud_min_z` setting and get rid of the other ground filter we wrote.

There are further changes I would make, but I think for now it's ok. Namely, this breaks if there are two different nodes that want to look at two different cross sections. The server only outputs one cross section, and it could very easily happen that two nodes publish z-height cross section requests that conflict with each other and the cross section quickly oscillates between the two. But - like I said, I think this is ok for now, given how we use it. Doing this would basically unify our modifications with the really really unfinished LayeredOctomapServer version.

**Comment by brenden-square** [comment]
@agdespopoulos We do have to concern ourselves with negative Z points because of the floor sumps in some tanks which are below floor level.  Check out the image below.  For our purposes, we want to avoid them just like any obstacle that lies above the tank floor, so my thought was to only exclude points in the envelope you're talking about.
efr_floor_sump

**Review by brenden-gibbons — APPROVED** [review]

**Comment by agdespopoulos** [comment]
Gotcha, makes sense!

**Review by agdespopoulos — APPROVED** [review]

**Pull request merged**